helix-core/src/main/java/org/apache/helix/task/TaskUtil.java

```diff
@@ -1037,6 +1037,53 @@ public static void purgeExpiredJobs(String workflow, WorkflowConfig workflowConf
     setNextJobPurgeTime(workflow, currentTime, purgeInterval, rebalanceScheduler, manager);
   }
 
+  /**
+   * The function that loops through the all existing workflow contexts and removes IdealState and
+   * workflow context of the workflow whose workflow config does not exist.
+   * @param workflowConfigMap
+   * @param resourceContextMap
+   * @param manager
+   */
+  public static void workflowGarbageCollection(final Map<String, WorkflowConfig> workflowConfigMap,
```

Comment:
Is this tool related to the TaskGarbageCollectionStage?
It would be better to split the PR if they are not strictly related.

Reply:
This is the logic I'm recovering with this PR; it's also the logic that @alirezazamani deleted in the linked PR in the description. :)

Reply:
You are not answering the question. Sorry but I don't have that context.
So is it related to the stage code, please?

Reply:
I see the confusion: this function is called on line 75 in the stage, please see above. 

I was trying to give you the context: this function was deleted in https://github.com/apache/helix/pull/803, and recovering it is the sole purpose of this PR.

Reply:
Got you.